test/JDBC/expected/replace-before-14_3-vu-verify.out

```diff
@@ -1,2855 +0,0 @@
-SELECT replace(@inputString, @pattern, @replacement)
-GO
-~~START~~
-nvarchar
-比尔·尔·比斯
-~~END~~
-
-
-DECLARE @inputString NVARCHAR(50) = N'比尔·拉莫斯', @pattern NVARCHAR(10) = N'拉莫', @replacement NVARCHAR(10) = N'尔·比'
-SELECT replace(@inputString, @pattern, @replacement) COLLATE CHINESE_PRC_CI_AS
-GO
-~~START~~
-nvarchar
-比尔·尔·比斯
-~~END~~
-
-
-DECLARE @inputString NVARCHAR(50) = N'比尔·拉莫斯', @pattern NVARCHAR(MAX) = N'拉莫', @replacement NVARCHAR(MAX) = N'尔·比'
-SELECT replace(@inputString, @pattern, @replacement)
-GO
-~~START~~
-nvarchar
-比尔·尔·比斯
-~~END~~
-
-
-DECLARE @inputString NVARCHAR(50) = N'比尔·拉莫斯', @pattern NVARCHAR(MAX) = N'拉莫', @replacement NVARCHAR(MAX) = N'尔·比'
-SELECT replace(@inputString, @pattern, @replacement) COLLATE CHINESE_PRC_CI_AS
-GO
-~~START~~
-nvarchar
-比尔·尔·比斯
-~~END~~
-
-
--- input type binary
-DECLARE @inputString BINARY(10) = 0x61626364656667
-SELECT replace(@inputString, 0x6364, 0x737475)
-GO
-~~START~~
-varchar
-abstuefg
-~~END~~
-
-
-DECLARE @inputString BINARY(10) = 0x61626364656667, @pattern BINARY(10) = 0x6364, @replacement BINARY(10) = 0x737475
-SELECT replace(@inputString, @pattern, @replacement)
-GO
-~~START~~
-varchar
-abstuefg
-~~END~~
-
-
--- with table column of type binary
-SELECT replace(a, b, c) FROM babel_4836_replace_t4
-GO
-~~START~~
-varchar
-atuvde
-~~END~~
-
-
--- input type varbinary
-DECLARE @inputString VARBINARY(10) = 0x61626364656667
-SELECT replace(@inputString, 0x6364, 0x737475)
-GO
-~~START~~
-varchar
-abstuefg
-~~END~~
-
-
-DECLARE @inputString VARBINARY(10) = 0x61626364656667, @pattern VARBINARY(10) = 0x6364, @replacement VARBINARY(10) = 0x737475
-SELECT replace(@inputString, @pattern, @replacement)
-GO
-~~START~~
-varchar
-abstuefg
-~~END~~
-
-
--- with table column of type varbinary
-SELECT replace(a, b, c) FROM babel_4836_replace_t5
-GO
-~~START~~
-varchar
-atuvde
-~~END~~
-
-
-
--- input type text
-SELECT replace(a, b, c) FROM babel_4836_replace_text
-GO
-~~START~~
-varchar
-  abcjhi?ghi??    
-~~END~~
-
-
-DECLARE @pattern VARCHAR(20) = '?de', @replacement VARCHAR(10) = 'x?y';
-SELECT replace(a, @pattern, @replacement) FROM babel_4836_replace_text
-GO
-~~START~~
-varchar
-  abc?defghi??    
-~~END~~
-
-
--- input type ntext
-SELECT replace(a, b, c) FROM babel_4836_replace_ntext
-GO
-~~START~~
-varchar
-  abcjhi?ghi??    
-~~END~~
-
-
-DECLARE @pattern NVARCHAR(20) = N'🙂de', @replacement NVARCHAR(10) = N'x🙂y';
-SELECT replace(a, @pattern, @replacement) FROM babel_4836_replace_ntext
-GO
-~~START~~
-nvarchar
-  abcx🙂yfghi🙂🙂    
-~~END~~
-
-
--- dependent objects
-SELECT * FROM babel_4836_replace_dep_view
-GO
-~~START~~
-text
-~~ERROR (Code: 33557097)~~
-
-~~ERROR (Message: nondeterministic collations are not supported for substring searches)~~
-
-
-SELECT * FROM babel_4836_replace_dep_view1
-GO
-~~START~~
-text
-~~ERROR (Code: 33557097)~~
-
-~~ERROR (Message: nondeterministic collations are not supported for substring searches)~~
```

Comment:
when we were testing these queries in upgrades coming from version before 14.3, to 15.latest, we were getting these errors which was not there for upgrades from version 14.3 and above to 15.latest. And this output difference was expected as before 14.3 we were using PG replace function for replace, which will throw error for non-deterministic collations. Hence we added these tests. Can you explain why we are not getting this error for upgrades from version before 14.3, to 16.latest ?

Reply:
We supported replace and charindex for non deterministic collation by putting hooks in postgres strpos() and replace() functions. We did not add a dialect check here because 
1. Dialect is set to PSQL when executing non pltsql functions. So dialect == TSQL would not work here.
2. We used ICU functions to implement this, hence replace() and strpos() will work even for postgres user defined non deterministic collation correctly.